src/main/java/com/microsoft/azure/vmagent/AzureVMCloud.java

```diff
@@ -132,7 +132,7 @@
     private List<AzureTagPair> cloudTags;
 
     //The map should not be accessed without acquiring a lock of the map
-    private final transient Map<AzureVMAgent, AtomicInteger> agentLocks = new HashMap<>();
+    private Map<AzureVMAgent, AtomicInteger> agentLocks = new HashMap<>();
```

Comment:
shouldn't this still be transient?

Reply:
I do not think so. 

But I do now know exactly when the plugin is serialized by Jenkins, I could not find any good documentation on this topic.
But i assuming that Jenkins would not sutdown a plugin and serialized it while there are active threads holding a lock or doing something.

But even it is is serialized while a job is pending and an object lock is held then after the deserialization these objects would be new objects and therefore a different lock and the locking mechanism would not work. 

This is why i removed the transient attribute, imho it does not make sense to serialize object locks.

Reply:
> This is why i removed the transient attribute, imho it does not make sense to serialize object locks.

then mark it as transient? transient means _do not serialise me_

this change means that there would be an additional map saved in the plugin configuration which I don't think is what we want. (the readResolve change looks correct)

Reply:
You are absolutly right.

It seemed I got confused while trying to understant the XStream framework. But of course we do not want to serialize it, therefor i will readd the transient keywork.